src/app/core/library-module/library.service.ts

```diff
@@ -165,27 +166,13 @@ export class LibraryService {
         })
       )
       .toPromise()
-      .then((response: HttpResponse<Blob>) => {
-        // Get the content disposition header from the response
-        const contentDisposition = response.headers.get('content-disposition');
-        // Get the blob from the response
-        const blob = response.body;
-        // Validate that the blob is not empty
-        if (!blob) {
-          throw this.handleError(new HttpErrorResponse({ error: 'No content in response body', status: 500 }));
+      .then((response: HttpResponse<{url: string}>) => {
```

Comment:
Can you just add a short description of how this is different from the current method? Thanks!